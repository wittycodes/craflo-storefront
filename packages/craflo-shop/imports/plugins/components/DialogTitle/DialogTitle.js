"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _DialogTitle = _interopRequireDefault(require("@material-ui/core/DialogTitle"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

/**
 * @name DialogTitle
 * @param {Object} props Component props
 * @returns {React.Component} A React component
 */
var DialogTitle = _react.default.forwardRef(function DialogTitle(props, ref) {
  var children = props.children,
      other = (0, _objectWithoutProperties2.default)(props, ["children"]);
  return /*#__PURE__*/_react.default.createElement(_DialogTitle.default, (0, _extends2.default)({
    disableTypography: true,
    ref: ref
  }, other), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    component: "h2",
    variant: "h4"
  }, children));
});

DialogTitle.propTypes = {
  /**
   * Children
   */
  children: _propTypes.default.node
};
var _default = DialogTitle;
exports.default = _default;