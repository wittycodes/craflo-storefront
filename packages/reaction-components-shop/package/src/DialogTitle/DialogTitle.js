import React from "react";
import PropTypes from "prop-types";
import MuiDialogTitle from "@material-ui/core/DialogTitle";
import Typography from "@material-ui/core/Typography";

/**
 * @name DialogTitle
 * @param {Object} props Component props
 * @returns {React.Component} A React component
 */
const DialogTitle = React.forwardRef(function DialogTitle(props, ref) {
  const { children, ...other } = props;

  return (
    <MuiDialogTitle
      disableTypography
      ref={ref}
      {...other}
    >
      <Typography component="h2" variant="h4">{children}</Typography>
    </MuiDialogTitle>
  );
});


DialogTitle.propTypes = {
  /**
   * Children
   */
  children: PropTypes.node
};

export default DialogTitle;
