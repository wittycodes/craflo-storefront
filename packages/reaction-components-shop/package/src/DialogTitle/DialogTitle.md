### Overview

The `DialogTitle` component provides a standard layout and typography styles for dialog titles. It uses [DialogTitle](https://material-ui.com/api/dialog-title) component from Material-UI.

### Usage

```jsx
<DialogTitle>Archive 24 products?</DialogTitle>
```

The `DialogTitle` component is to be used within body of the [Dialog](https://material-ui.com/api/dialog) component.

```jsx static
import Dialog from "@material-ui/core/Dialog";

<Dialog>
  <DialogTitle>Archive 24 products?</DialogTitle>
</Dialog>
```
