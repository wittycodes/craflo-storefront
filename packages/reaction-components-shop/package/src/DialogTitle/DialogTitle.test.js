import React from "react";
import { render } from "../tests/index.js";
import DialogTitle from "./DialogTitle";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<DialogTitle>Archive 24 products?</DialogTitle>);
  expect(asFragment()).toMatchSnapshot();
});
