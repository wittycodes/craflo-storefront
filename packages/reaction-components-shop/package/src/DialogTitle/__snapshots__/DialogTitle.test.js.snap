// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <div
    class="MuiDialogTitle-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h4"
    >
      Archive 24 products?
    </h2>
  </div>
</DocumentFragment>
`;
